pkg/test/framework/components/rawvm/gce.go

```diff
@@ -0,0 +1,57 @@
+// Package rawvm implements features for Istio mesh expansion.
+package rawvm
+
+import (
+	"fmt"
+
+	old_framework "istio.io/istio/tests/e2e/framework"
+
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+// gceComponent is the mesh expansion VM implementation on GCP.
+type gceComponent struct {
+	// id is the GCE instance id.
+	id    resource.ID
+	rawVM *old_framework.GCPRawVM
+}
+
+// NewGCE creates a GCE instance that finishes the setup with specified application.
+func NewGCE(ctx resource.Context, config Config) (Instance, error) {
```

Comment:
I'm not sure this is the right way to organize this. It sounds like you're effectively trying to define 2 new environments here: GCE and RawVM. Assuming you've filled out the components for both environments, you could theoretically run existing tests on those environments.

If the need here is to combine vms and k8s in the same test, would it be reasonable to consider that a multi-cluster configuration? We don't currently support multi-cluster in the test framework API surface, but we'll probably have to at some point.

cc @smawson  @louiscryan @ozevren